networkx/algorithms/hierarchy.py

```diff
@@ -42,6 +49,9 @@ def flow_hierarchy(G, weight=None):
        DOI: 10.1002/cplx.20368
        http://web.mit.edu/~cmagee/www/documents/28-DetectingEvolvingPatterns_FlowHierarchy.pdf
     """
+    # corner case: G has no edges
+    if nx.is_empty(G):
+        raise nx.NetworkXPointlessConcept("G must have edges in flow_hierarchy")
```

Comment:
```suggestion
    # corner case: G has no edges
    if nx.is_empty(G):
        raise nx.NetworkXError("flow_hierarchy not applicable to empty graphs")
```

Reply:
so in such cases, we should adopt the `NetworkXError` ?

Reply:
`NetworkXPointlessConcept` is for null graph(0 nodes 0 edges) cases and here we are checking if the graph is empty(i.e n nodes 0 edges). So, that's why we switched back to `NetworkXError`. Sorry for the confusion!